analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyNullableValueAccess.cs

```diff
@@ -737,6 +737,46 @@ void RefParams(int? iParam)
     }
 }
 
+class InParams
+{
+    void InParamOfALocalFunction(int? iParam)
```

Comment:
This test and the `LocalAsInParamToAnotherMethod` test are almost identical. I would remove the local function here and use `MethodWithReadInParam` instead. LocalAsInParamToAnotherMethod can be removed then. `MethodWithReadInParam` can be named InParam or something like that. I don't see how "Method" "With" and "Read" help here. You should instead add the optional `in` modifier on the call side: `InParam(in iParam)`. That makes it obvious that the test is about the modifier's semantics.

Reply:
Add the optional `in` modifier on the call side: done.

Remove the local function and use `MethodWithReadInParam`: done. The idea here was to test both local functions and class instance methods, and not make any assumption on whether the underlying symbolic execution engine treats the two structures differently. I guess there would or will be differences in how inter-procedural execution is done, across methods, non-static local functions and static local functions. 

`MethodWithReadInParam` rename into `InParam`: done.